codemark/firebase/database.py

```diff
@@ -81,6 +81,29 @@ def signout(self):
         except Exception as e:
             print_error(e)
 
+    def register(self, email, password):
+        '''
+        Registers a user to the firebase realtime database 
+        :param email: Email
+        :param password: Password
+        '''
+        try:
```

Comment:
Try following this API for possible implementations, https://github.com/rootCircle/KANS/blob/f8174cb4f9857adc750ac79fb35ac792ef2708d1/App.py#L602

Current one leaves few edge cases for `Email Already Exists`, `Malformed Email`. I appreciate that you kept that connection checking on board.

You can modify this code as per your need to change them. (Yours uses `pyrebase` library, below one use `firebase-admin`)
```py
def register(self, email, password, username, details, table):
        """
        Provide subscrption to authentication as well as creates another table for storing other details in
        Realtime databse simultaneously.
        :param email : Email
        :param password: Password
        :param username: Username(Used to associate email with username without creating separate table)
        :param details: Used to fetch Name to put it in Firebase authentication, Also to put all other data plus email
                        and username in separate RealTime Database
        :param table: Name of the table in which the details would be put in.
        :return: True : If everything is Ok
        None : If some error arises, it is often accompanied by a prompt displaying common errors like Duplicate Email,
                Invalid mail format, Weak Password etc. It also returns None if it can't connect to FirebaseDB
        """
        try:
            user = auth.create_user(email=email, password=password, display_name=details['Name'])
            data = {"Username": username, "Email": email}
            data.update(details)
            results = db.reference(table).push(data)
            return True
        except requests.exceptions.ConnectionError:
            # Handle the "Network is unreachable" error
            print_error("Network is unreachable.")
        except (ValueError, fireexception.FirebaseError, fireexception.UnavailableError) as e:
            e = str(e)
            if e.find("EMAIL_EXISTS") != -1:
                print("Account with this email already created",
                                    "Account with this email already created.\nTry Another Email Address")
            elif e.find("Invalid password") != -1:
               print("Weak Password", "Password must be of atleast 6 characters.")
            elif e.find("Malformed email address") != -1:
                print("Invalid Email Address", "Email Address must be of Valid format!")
            else:
                print("Error", e)
```

Reply:
The one in `signup.py` is a client side checking, and one we're are aiming are server side handling, both has a difference. [Both are important btw]

Reason :  Maybe our checking might be flawed (it certainly can be), then in case our checks allows email to pass through the servers, which server will automatically detect it out(thanks to firebase), and hence will throw an error. But, since we won't be handling that error, means ugly code at the end :->

Reply:
I will wait for the Firebase_admin one to be implemented before closing this! If you're having any trouble in that, then I can certainly help you out!

Reply:
Oh ok should I change the pyrebase implementation to firebase_admin one? I was following how the `login` function was implemented in `database.py` at L113

Reply:
Yes! exactly! Login one implementation will be changed in future! 

For now, you can follow the code above for reference!